src/libstd/io/mod.rs

```diff
@@ -566,21 +1008,40 @@ pub trait BufRead: Read {
     ///
     /// If an I/O error is encountered then all bytes read so far will be
     /// present in `buf` and its length will have been adjusted appropriately.
+    ///
+    /// # Examples
+    ///
+    /// A locked standard input implements `BufRead`. In this example, we'll
+    /// read from standard input until we see an `a` byte.
+    ///
+    /// ```
+    /// use std::io;
+    /// use std::io::BufRead;
+    ///
+    /// let stdin = io::stdin();
+    /// let mut stdin = stdin.lock();
+    /// let mut buffer = Vec::new();
+    ///
+    /// // casting a char to a u8 is nicer than writing 97
+    /// stdin.read_until('a' as u8, &mut buffer).unwrap();
```

Comment:
It avoids the need for the comment about writing 97 and I'd consider it the "more idiomatic" way to do it